drivers/HwWatchdog.cpp

```diff
@@ -0,0 +1,100 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "HwWatchdog.h"
+
+static LowPowerTicker ticker;//tiker for periodic delay call back for kick()
```

Comment:
A singleton pointer or getter function should be used here to ensure this does not take up RAM when it is not used. Something like
```C++
SingletonPtr<LowPowerTicker> ticker;
```

or
```C++
static LowPowerTicker *get_ticker()
{
    static LowPowerTicker ticker;
    return &ticker;
}
```